utils/lib/rr_graph_capnp/graph2.py

```diff
@@ -0,0 +1,548 @@
+            )
+
+            if switch.timing:
+                timing = out_switch.timing
+                timing.cin = switch.timing.c_in
+                timing.cinternal = switch.timing.c_internal
+                timing.cout = switch.timing.c_out
+                timing.r = switch.timing.r
+                timing.tdel = switch.timing.t_del
+                timing.penaltyCost = switch.timing.p_cost
+
+            if switch.sizing:
+                sizing = out_switch.sizing
+                sizing.bufSize = switch.sizing.buf_size
+                sizing.muxTransSize = switch.sizing.mux_trans_size
+
+    def _write_segments(self, rr_graph):
+        """
+        Writes the RR graph segments.
+        """
+
+        segments = rr_graph.segments.init('segments', len(self.graph.segments))
+
+        for out_segment, segment in zip(segments, self.graph.segments):
+            out_segment.id = segment.id
+            out_segment.name = segment.name
+
+            if segment.timing:
+                timing = out_segment.timing
+                timing.cPerMeter = segment.timing.c_per_meter
+                timing.rPerMeter = segment.timing.r_per_meter
+
+    def _write_block_types(self, rr_graph):
+        """
+        Writes the RR graph block types.
+        """
+
+        block_types = rr_graph.blockTypes.init(
+            'blockTypes', len(self.graph.block_types)
+        )
+
+        for out_blk, blk in zip(block_types, self.graph.block_types):
+            out_blk.id = blk.id
+            out_blk.name = blk.name
+            out_blk.width = blk.width
+            out_blk.height = blk.height
+
+            pin_classes = out_blk.init('pinClasses', len(blk.pin_class))
+
+            for out_pin_class, pin_class in zip(pin_classes, blk.pin_class):
+                out_pin_class.type = to_capnp_enum(
+                    self.rr_graph_schema.PinType, pin_class.type
+                )
+
+                pins = out_pin_class.init('pins', len(pin_class.pin))
+
+                for out_pin, pin in zip(pins, pin_class.pin):
+                    out_pin.ptc = pin.ptc
+                    out_pin.value = pin.name
+
+    def _write_grid(self, rr_graph):
+        """
+        Writes the RR graph grid.
+        """
+
+        grid_locs = rr_graph.grid.init('gridLocs', len(self.graph.grid))
+        for out_grid_loc, grid_loc in zip(grid_locs, self.graph.grid):
+            out_grid_loc.x = grid_loc.x
+            out_grid_loc.y = grid_loc.y
+            out_grid_loc.blockTypeId = grid_loc.block_type_id
+            out_grid_loc.widthOffset = grid_loc.width_offset
+            out_grid_loc.heightOffset = grid_loc.height_offset
+
+    def serialize_to_capnp(
+            self,
+            channels_obj,
+            connection_box_obj,
+            num_nodes,
+            nodes_obj,
+            num_edges,
+            edges_obj,
+            node_remap=lambda x: x
+    ):
+        """
+        Writes the routing graph to the XML file.
```

Comment:
This is not an XML file anymore right? I mean, it gets binarized by capnp, so maybe this comment could be misleading

Reply:
Fixed